src/Tools/dotnet-monitor/CollectionRules/Options/CollectionRuleOptions.Validate.cs

```diff
@@ -15,6 +15,13 @@ IEnumerable<ValidationResult> IValidatableObject.Validate(ValidationContext vali
         {
             List<ValidationResult> results = new();
 
+            results.AddRange(ErrorList);
+
+            if (results.Count > 0)
+            {
+                return results; // QUESTION: Do we want to short-circuit here? Invalid template names may cause other error messages that distract from the actual issue (e.g. saying the Type field is required on a trigger when the trigger template wasn't found)
```

Comment:
I think as long as we show the error that the template is invalid, further validation makes no sense

Reply:
My thinking on this is that it makes more sense to short-circuit than to show error-messages that might be misleading (as the comment suggests). Technically, we could probably add in the logic to recognize that, for example, if the trigger template failed, we don't do the validation on the trigger (but to do the other pieces), but I also don't think we want/need to be adding additional complexity just for this. We do already have a precedent for this (kind of) - in `CollectionRuleOptionsTests`, there's a comment noting that validation short-circuits before checking `GreaterThan` and `LessThan` for `EventCounterTrigger`. Once the user has fixed their templates, they'll still see the errors - fixing config just might require an extra iteration, which doesn't _seem_ unreasonable.

Reply:
As a user, it annoys me when VS tells me "you have 3 errors", I fix them, and then it says "you now have 10 errors", fix them, etc, until some indeterminate point in the future when I have no errors left. I would understand that if fixing some error uncovers other related errors, but errors that are independent of each other should be reported at the same time.

This is also something that could be fixed later.